### PR TITLE
Support Face Unlock

### PR DESCRIPTION
**Pull request by brandonpage**
Add support for Face Unlock and Iris.  UX cleanup and fixed warnings.

Missing: Icon for Face Unlock enrollment screen.

### CODE REVIEW

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/PasscodeActivity.java [hunk 1]
```diff
@@ -491,6 +536,18 @@ public void onAuthenticationFailed() {
         }
     }
 
+    @TargetApi(VERSION_CODES.Q)
```
Comment (brandonpage) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/PasscodeActivity.java]: This is the main point of the PR.  Biometric Manager is new in API 29.

File: native/NativeSampleApps/test/RestExplorerTest/src/com/salesforce/samples/restexplorer/PasscodeActivityTest.java [hunk 2]
```diff
@@ -761,7 +761,7 @@ private void checkUi() {
                 Assert.assertEquals("Expected biometric title instructions to be visible.", View.VISIBLE,
                         passcodeActivity.findViewById(R.id.sf__biometric_instructions_title).getVisibility());
                 Assert.assertEquals("Biometric instructions title is wrong.",
-                        passcodeActivity.getString(R.string.sf__biometric_allow_instructions_title),
+                        passcodeActivity.getString(R.string.sf__biometric_fingerprint_allow_instructions_title),
```
Comment (brandonpage) [inline native/NativeSampleApps/test/RestExplorerTest/src/com/salesforce/samples/restexplorer/PasscodeActivityTest.java]: Test fix.

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/PasscodeActivity.java [hunk 3]
```diff
@@ -188,6 +197,15 @@ public void onClick(View v) {
                 passcodeField.setText(inputText.trim());
             }
         }
+
+        // Determine biometric hardware, default is fingerprint.
+        if (VERSION.SDK_INT >= VERSION_CODES.Q) {
```
Comment (bhariharan) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/PasscodeActivity.java]: Add a `TODO:` here to remove this check once `minVersion > API 29`.

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/PasscodeActivity.java [hunk 4]
```diff
@@ -188,6 +197,15 @@ public void onClick(View v) {
...
+            if (getPackageManager().hasSystemFeature(PackageManager.FEATURE_FACE)) {
```
Comment (wmathurin) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/PasscodeActivity.java]: Can Face and Iris unlock be available on the same device?

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/PasscodeActivity.java (hunk 1 above)
Comment (bhariharan) [reply libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/PasscodeActivity.java]: Add a `TODO:` here to remove this conditional if `minApi >= 29`.

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/PasscodeActivity.java (hunk 4 above)
Comment (brandonpage) [reply libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/PasscodeActivity.java]: I don't believe so.  We can decide if we want to update the logic to allow choice or something in the future _if_ a device is ever release that has more than 1 of the 3.

**Pull request merged**